### ISSUE TITLE
Mouse scroll breaks visual selection

### ISSUE THREAD

**Issue by pudiva** [vim/vim#15710]
### Steps to reproduce

1. open a terminal (tested on alacritty, kitty, foot and st)
2. run vim
3. type a bunch of stuff so that you have many lines to scroll
4. start visual selection by pressing left mouse button and moving cursor (don't release yet!)
5. scroll mouse wheel
6. move the mouse cursor and see the page scrolling up whenever you move

See screen capture:

https://github.com/user-attachments/assets/2240b15f-d658-4c83-99ac-5016bfce2d16

### Expected behaviour

mouse scrolling during a visual selection shouldn't cause subsequent mouse pointer movements to scroll and shouldn't stop stop them from extending or shrinking the visual selection.

### Version of Vim

9.1.0682

### Environment

Operating system: void-linux
Terminal: alacritty, kitty, foot and st
Value of $TERM: alacritty
Shell: /bin/bash

**Comment by chrisbra** [comment]
am i the only one, who cannot watch the video? It says corrupted

**Comment by pudiva** [comment]
> am i the only one, who cannot watch the video? It says corrupted

you are right... it did play on my computer after i uploaded it, but now on my phone it's totally glitched...

i'm travelling now but i'll try to upload again tomorrow when i arrive.

**Comment by pudiva** [comment]
@chrisbra I reencoded the video with some compatibility settings and updated the issue and now it works on my phone. could you take a look and confirm?

**Comment by pudiva** [comment]
interestingly the video now doesn't play on my computer... 🫠

**Comment by pudiva** [comment]
ok I think it had something to do with framerate. now it plays everywhere for me.

**Comment by warp-records** [comment]
Hey, I can't reproduce this error. I'm on Mac OS and used altra kitty with vim 9.1.

**Comment by pudiva** [comment]
I never tried it on Mac OS. This was recorded on void-linux.

**Comment by fybits** [comment]
I have exactly the same issue in kitty, alacritty and Konsole. I'm on cachyos. Works fine in neovim
Vim 9.2 1-81
